suite/tests/linux/syscall_pwait.c

```diff
@@ -161,17 +185,50 @@ main(int argc, char *argv[])
         /* XXX i#3240: DR currently does not handle the atomicity aspect of this system
          * call. Once it does, please include this in this test or add a new test.
          */
-        sigset_t *test_set_check = &test_set;
+        sigset_t pre_syscall_set;
+        sigprocmask(SIG_SETMASK, NULL, &pre_syscall_set);
         if (ppoll(NULL, 0, NULL, &test_set) == -1) {
             if (errno != EINTR)
                 perror("expected EINTR");
-            if (&test_set != test_set_check)
-                perror("sigmask mismatch");
         } else {
             perror("expected interruption of syscall");
         }
+        sigset_t post_syscall_set;
+        sigprocmask(SIG_SETMASK, NULL, &post_syscall_set);
+        if (!compare_sigmasks(&pre_syscall_set, &post_syscall_set))
+            perror("sigmask mismatch");
     }
 
+    print("Testing epoll_pwait failure\n");
+
+    /* XXX: The following failure tests will 'hang' if syscall succeeds, due to the nature
+     * of the syscall. Maybe change this into something that will rather fail immediately.
+     */
+
+    /* waste some time */
+    nanosleep(&sleeptime, NULL);
+
+    if (syscall(SYS_epoll_pwait, epoll_fd, &events, 24LL, -1LL, &test_set, 0) == 0)
+        perror("expected syscall failure");
```

Comment:
But returning 0 is success, and errno won't be set, so perror doesn't make sense here: it's going to print some unrelated earlier errno or sthg.  Ditto below.